xmanager/xm_local/storage/database.py

```diff
@@ -239,36 +237,46 @@ def insert_experiment(
         'INSERT INTO experiment (experiment_id, experiment_title) '
         'VALUES (:experiment_id, :experiment_title)'
     )
-    self.engine.execute(
-        query, experiment_id=experiment_id, experiment_title=experiment_title
-    )
+    with self.engine.begin() as connection:
+      connection.execute(
+          query,
+          {
+              'experiment_id': experiment_id,
+              'experiment_title': experiment_title,
+          },
+      )
 
   def insert_work_unit(self, experiment_id: int, work_unit_id: int) -> None:
     query = text(
         'INSERT INTO work_unit (experiment_id, work_unit_id) '
         'VALUES (:experiment_id, :work_unit_id)'
     )
-    self.engine.execute(
-        query, experiment_id=experiment_id, work_unit_id=work_unit_id
-    )
+    with self.engine.begin() as connection:
+      connection.execute(
+          query,
+          {'experiment_id': experiment_id, 'work_unit_id': work_unit_id},
+      )
 
   def insert_vertex_job(
       self, experiment_id: int, work_unit_id: int, vertex_job_id: str
   ) -> None:
     job = data_pb2.Job(caip=data_pb2.AIPlatformJob(resource_name=vertex_job_id))
-    data = text_format.MessageToBytes(job)
+    data = text_format.MessageToString(job)
```

Comment:
This seemed like a bug to me. The database column for `job_data` is of type `sa.String(255)`, so I think the data should be formatted as string and not bytes. `MessageToString ` is also in line with what `insert_kubernetes_job` below does already.